semgrep/semgrep/commands/app_scan.py

```diff
@@ -0,0 +1,122 @@
+import sys
+import time
+from pathlib import Path
+from typing import Any
+from typing import Iterable
+from typing import Sequence
+
+import click
+
+from semgrep.commands.login import Authentication
+from semgrep.commands.scan import CONTEXT_SETTINGS
+from semgrep.commands.scan import scan
+from semgrep.commands.scan import scan_options
+from semgrep.error import FATAL_EXIT_CODE
+from semgrep.error import SemgrepError
+from semgrep.ignores import IGNORE_FILE_NAME
+from semgrep.semgrep_app import ScanHandler
+
+# These patterns are excluded via --exclude regardless of other ignore configuration
+ALWAYS_EXCLUDE_PATTERNS = [".semgrep/", ".semgrep_logs/"]
+
+# These patterns are excluded via --exclude unless the user provides their own .semgrepignore
+DEFAULT_EXCLUDE_PATTERNS = ["test/", "tests/", "*_test.go"]
+
+
+def yield_valid_patterns(patterns: Iterable[str]) -> Iterable[str]:
+    for pattern in patterns:
+        pattern = pattern.strip()
+
+        if not pattern:
+            continue
+        if pattern.startswith("#"):
+            continue
+
+        yield pattern
+
+
+def yield_exclude_args(requested_patterns: Sequence[str]) -> Iterable[str]:
+    patterns = [*yield_valid_patterns(requested_patterns), *ALWAYS_EXCLUDE_PATTERNS]
+    if Path(IGNORE_FILE_NAME).is_file():
+        patterns.extend(DEFAULT_EXCLUDE_PATTERNS)
+
+    for pattern in patterns:
+        yield from ["--exclude", pattern]
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.pass_context
+@scan_options
+def app_scan(ctx: click.Context, *args: Any, **kwargs: Any) -> None:
+    token = Authentication.get_token()
+    if not token:
+        click.echo("run `semgrep login` before using `semgrep ci`", err=True)
+        sys.exit(1)
+    if not Authentication.is_valid_token(token):
+        click.echo(
+            "API token not valid. Try to run `semgrep logout` and `semgrep login` again.",
+            err=True,
+        )
+        sys.exit(1)
+
+    scan_handler = ScanHandler(token)
+    metadata = {"repository": "returntocorp/semgrep"}
```

Comment:
left here accidentally?

Reply:
Placeholder for now while I work on https://github.com/returntocorp/semgrep/pull/4820